### ISSUE TITLE
Spring Cloud GCP Starters in start.spring.io

### ISSUE THREAD

**Issue by saturnism** [spring-cloud/spring-cloud-gcp#181]
Add the starters to start.spring.io

**Comment by joaoandremartins** [comment]
The suggested list of starters is:

1. `spring-cloud-gcp-starter-core` - Provides core functionality like GCP project ID and credentials providers.
2. `spring-cloud-gcp-starter-config` - Enables Spring Cloud GCP Runtime Configuration.
3. `spring-cloud-gcp-starter-logging` - Enables Spring Cloud GCP Logging.
4. `spring-cloud-gcp-starter-pubsub` - Enables Spring Cloud GCP Pub/Sub. Auto-configures the Pub/Sub bits of `spring-cloud-gcp-integration` and `spring-cloud-gcp-stream`.
5. `spring-cloud-gcp-starter-sql` - Enables Spring Cloud GCP JDBC for Cloud SQL.
6. `spring-cloud-gcp-starter-storage` - Enables Spring Resources on Cloud Storage.
7. `spring-cloud-gcp-starter-trace` - Enables Spring Cloud Sleuth through Stackdriver Trace.

This is similar to the list proposed in https://github.com/spring-io/initializr/pull/539

@snicoll thoughts?

**Comment by snicoll** [comment]
> @snicoll thoughts?

Yes, we should finalize the issues that I've raised before discussing this. I've already shared that we won't add all of them.

**Comment by meltsufin** [comment]
Currently, `spring-cloud-gcp-starter-core` is an alias for `spring-cloud-gcp-starter`. How would you want to see it changed?

**Comment by joaoandremartins** [comment]
@snicoll I was hoping we could get to a consensus sooner rather than later.

As I see it, the starters that will be added to Initializr don't depend on the list of issues from https://github.com/spring-io/initializr/pull/539#issuecomment-352026279, so I thought we could have this conversation in parallel, but let me know if I'm missing anything.

It sounds like we could benefit from a VC meeting to discuss this. @meltsufin wdyt? Should we freeze some time later this week/earlier next week?

**Comment by snicoll** [comment]
I am happy to discuss but the whole reason of the issues I've created is to come up with a situation where the code is ready to be consumed on start.spring.io. My proposal of having a base starter is key in this process, as [I've indicated several times already](https://github.com/spring-cloud/spring-cloud-gcp/issues/262#issuecomment-352169157).

I'd rather wait for that state to happen first.

**Comment by snicoll** [comment]
>  Currently, spring-cloud-gcp-starter-core is an alias for spring-cloud-gcp-starter. How would you want to see it changed?

No it isn't. My proposal imply that the auto-configuration code will be made available and the starters have no code. We're not there yet.

**Comment by meltsufin** [comment]
@snicoll We're getting there with #276, which will be followed by another PR to make all starters have no code. #276 only does this for the `config` module, but it's a template for all others to follow next.

**Comment by joaoandremartins** [comment]
We already have starters in Initializr.
This issue can now track adding more starters to Initializr.

@snicoll what are you first thoughts on adding the remaining Spring Cloud GCP starters to Initializr?

**Comment by snicoll** [comment]
I think we should first document how to use start.spring.io to integrate with all the features that have been developed (we took the time to create a GCP support entry, I'd like us to document how users can use it). We also need to wait a bit and see how popular it is before considering adding more entries.

**Comment by meltsufin** [comment]
@snicoll Where do you think the documentation for start.spring.io go? Our refdoc?

**Comment by snicoll** [comment]
Anything that the GCP community can easily find. Can be on the github project, in the ref guide. If you have links we can add them in the metadata so I think that's quite interesting as well.

**Comment by joaoandremartins** [comment]
I'll close here since we have the docs issue ready in #598.
